gradle/android-butterknife/app/src/main/java/org/example/kotlin/butterknife/SimpleAdapter.kt

```diff
@@ -0,0 +1,52 @@
+package org.example.kotlin.butterknife
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseAdapter
+import android.widget.TextView
+import butterknife.ButterKnife
+import butterknife.Bind
+
+public class SimpleAdapter(context: Context) : BaseAdapter() {
+
+    private val inflater = LayoutInflater.from(context)
+
+    override fun getCount() = CONTENTS.size()
+    override fun getItem(position: Int) = CONTENTS[position]
+    override fun getItemId(position: Int) = position.toLong()
+
+    override fun getView(position: Int, v: View?, parent: ViewGroup): View {
+        var view = v
+        val holder: ViewHolder
+        if (view != null) {
+            holder = view.tag as ViewHolder
+        } else {
+            view = inflater.inflate(R.layout.simple_list_item, parent, false)
+            holder = ViewHolder(view)
+            view!!.tag = holder
+        }
+
+        val word = getItem(position)
+        holder.word.text = "Word: $word"
+        holder.length.text = "Length: ${word.length()}"
+        holder.position.text = "Position: $position"
+
+        return view
+    }
+
+    class ViewHolder(view: View) {
+        @Bind(R.id.word) lateinit val word: TextView
```

Comment:
Different style in different places: let's put annotations on the previous line everywhere